src/app/(sidebar)/account/fund/page.tsx

```diff
@@ -1,5 +1,185 @@
 "use client";
 
+import { useState } from "react";
+import Link from "next/link";
+
+import {
+  Card,
+  Icon,
+  IconButton,
+  Input,
+  Text,
+  Button,
+} from "@stellar/design-system";
+
+import { shortenStellarAddress } from "@/helpers/shortenStellarAddress";
+import { validatePublicKey } from "@/helpers/validatePublicKey";
+import { useStore } from "@/store/useStore";
+
+import { AlertBox } from "@/components/AlertBox";
+
+import "../styles.scss";
+
+const callFriendBot = async ({
+  network,
+  publicKey,
+}: {
+  network: string;
+  publicKey: string;
+}) => {
+  const friendbotURL =
+    network === "futurenet"
+      ? "https://friendbot-futurenet.stellar.org"
+      : "https://friendbot.stellar.org";
+  const response = await fetch(friendbotURL + "/?addr=" + publicKey);
+
+  return response;
+};
+
 export default function FundAccount() {
-  return <div>Fund Account</div>;
+  const { account, network } = useStore();
+
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const [generatedPublicKey, setGeneratedPublicKey] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const fundAccount = async () => {
+    try {
+      setIsLoading(true);
+      setError("");
+
+      const data = await callFriendBot({
+        network: network.id,
+        publicKey: account.publicKey,
+      });
+
+      if (!data.ok) {
+        const errorBody = await data.json();
+
+        throw new Error(errorBody.status || "An error occurred");
+      }
+
+      setIsSuccess(true);
+    } catch (error: any) {
+      let errorMessage = "";
+
+      if (error.status === 0) {
+        errorMessage = `Unable to reach Friendbot server at ${network.id}`;
+      } else {
+        errorMessage = `Failed to fund ${account.publicKey} on the ${network.id}`;
+      }
+      setError(errorMessage);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <div className="Account">
```

Comment:
To set the container's spacing, we'll need to add a gap to this class. Where possible, let's not use margin/padding on individual elements for spacing.